File: MarriageCalculator/MarriageCalculator.Core/Models/MarriageGameSet.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MarriageCalculator.Core.Models;

public class MarriageGameSet
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    public string HostUserId { get; set; } = string.Empty; // Creator/Host of this game set

    public string Name { get; set; }

    public DateTime LastPlayed { get; set; }
    public DateTime Created { get; set; }
    public bool IsActive { get; set; } = true;

    [BsonRepresentation(BsonType.ObjectId)]
    public string GameSettingsId { get; set; } = string.Empty;

    [BsonIgnore]
    public GameSettings GameSettings { get; set; } = GameSettings.Default();

    [BsonIgnore]
    public Dictionary<string, MarriageGameSetPlayer> GameSetPlayers { get; set; } = [];

    [BsonIgnore]
    public List<MarriageGameRound> Rounds { get; set; } = [];

    public MarriageGameSet()
    {
        Name = $"{DateTime.Now:yyyyMMdd HHmmss}";
        Created = DateTime.Now;
    }
}

File: MarriageCalculator/Android/app/src/main/java/com/sanjeeb/marriagecalculator/data/model/MarriageGameSet.kt
package com.sanjeeb.marriagecalculator.data.model

import com.google.gson.annotations.SerializedName

data class MarriageGameSet(
    @SerializedName("id") val id: String = "",
    @SerializedName("name") val name: String = "",
    @SerializedName("lastPlayed") val lastPlayed: String = "",
    @SerializedName("created") val created: String = "",
    @SerializedName("isActive") val isActive: Boolean = true,
    @SerializedName("gameSettingsId") val gameSettingsId: String = "",
    @SerializedName("gameSettings") val gameSettings: GameSettings? = null,
    @SerializedName("gameSetPlayers") val gameSetPlayers: Map<String, MarriageGameSetPlayer>? = null,
    @SerializedName("rounds") val rounds: List<MarriageGameRound>? = null
)

data class MarriageGameSetPlayer(
    @SerializedName("id") val id: String = "",
    @SerializedName("marriageGameSetId") val marriageGameSetId: String = "",
    @SerializedName("playerId") val playerId: String = "",
    @SerializedName("position") val position: Int = 0,
    @SerializedName("isActive") val isActive: Boolean = true,
    @SerializedName("player") val player: Player? = null
)

data class CreateGameSetRequest(
    @SerializedName("name") val name: String,
    @SerializedName("gameSettingsId") val gameSettingsId: String,
    @SerializedName("playerIds") val playerIds: List<String> = emptyList()
)
